frontend/public/components/overview/project-overview.tsx

```diff
@@ -143,80 +150,84 @@ const ProjectOverviewListItem = ({item, metrics, onClick, selectedItem}) => {
   </div>;
 
   return <ListView.Item
-    onClick={() => isSelected ? onClick({}) : onClick(item)}
+    onClick={() => store.dispatch(UIActions.selectOverviewItem(isSelected ? '' : uid))}
     className={className}
     heading={heading}
     additionalInfo={[additionalInfo]}
   />;
 };
 
-ProjectOverviewListItem.displayName = 'ProjectOverviewListItem';
-
-ProjectOverviewListItem.propTypes = {
-  item: PropTypes.shape({
-    controller: PropTypes.object,
-    obj: PropTypes.object.isRequired,
-    readiness: PropTypes.object,
-  }).isRequired
-};
-
-const ProjectOverviewList = ({items, metrics, onClickItem, selectedItem}) => {
+const ProjectOverviewList: React.SFC<ProjectOverviewListProps> = ({items, metrics, selectedUID}) => {
   const listItems = _.map(items, (item) =>
     <ProjectOverviewListItem
-      key={item.obj.metadata.uid}
       item={item}
+      key={item.obj.metadata.uid}
       metrics={metrics}
-      onClick={onClickItem}
-      selectedItem={selectedItem}
+      selectedUID={selectedUID}
     />
   );
   return <ListView className="project-overview__list">
     {listItems}
   </ListView>;
 };
 
-
-ProjectOverviewList.displayName = 'ProjectOverviewList';
-
-ProjectOverviewList.propTypes = {
-  items: PropTypes.array.isRequired
-};
-
-const ProjectOverviewGroup = ({heading, items, metrics, onClickItem, selectedItem}) =>
+const ProjectOverviewGroup: React.SFC<ProjectOverviewGroupProps> = ({heading, items, metrics, selectedUID}) =>
   <div className="project-overview__group">
     {heading && <h2 className="project-overview__group-heading">{heading}</h2>}
-    <ProjectOverviewList items={items} metrics={metrics} onClickItem={onClickItem} selectedItem={selectedItem} />
+    <ProjectOverviewList
+      items={items}
+      metrics={metrics}
+      selectedUID={selectedUID}
+    />
   </div>;
 
-
-ProjectOverviewGroup.displayName = 'ProjectOverviewGroup';
-
-ProjectOverviewGroup.propTypes = {
-  heading: PropTypes.string,
-  items: PropTypes.array.isRequired
+const stateToProps = ({UI}, ownProps): ProjectOverviewProps => {
+  return {
+    ...ownProps,
+    selectedUID: UI.getIn(['overview', 'selectedUID'])
+  };
 };
 
-export const ProjectOverview = ({selectedItem, groups, metrics, onClickItem}) =>
+export const ProjectOverview = connect(stateToProps)(({groups, metrics, selectedUID}) =>
   <div className="project-overview">
     {_.map(groups, ({name, items, index}) =>
       <ProjectOverviewGroup
         key={name || `_${index}`}
         heading={name}
         items={items}
         metrics={metrics}
-        onClickItem={onClickItem}
-        selectedItem={selectedItem}
+        selectedUID={selectedUID}
```

Comment:
I wonder if children shouldn't just connect to get selected UID from redux. (I don't know best practices here.)

Reply:
From what I understand, best practice is that the container component should be connected and pass down props. If we connect the child component, then every item in the list is connecting to redux. It's probably not that expensive, but it makes sense to just connect in one place and pass that prop down.

Reply:
There seems to be some debate on this. From what I'm reading, performance is probably better if you connect children instead of passing props.

I think we should definitely prefer connect if we're passing props through several layers of components that aren't directly using the props, just passing them down. I'd have to revisit the code to see if that's the case here.

https://stackoverflow.com/questions/47300361/use-connect-or-pass-data-as-props-to-children

Reply:
Yeah, I think it is the case. The `selectedUID` prop is being passed through `ProjectOverview` > `ProjectOverviewGroup` > `ProjectOverviewList` and is finally used in `ProjectOverviewListItem`